fix ggproto function and added super class to scale function

vivekktiwari opened the pull request.

vivekktiwari committed:
Merge pull request #3 from vivekktiwari/validate-params

adding few documentation

vivekktiwari committed:
updating dpendency package scales version to 0.4.1

vivekktiwari committed:
change from - validate-params to - cran

vivekktiwari committed:
Update .travis.yml

vivekktiwari committed:
Update DESCRIPTION

vivekktiwari committed:
testing

vivekktiwari committed:
fetching super class as function

vivekktiwari committed:
adding super class to all scale functions

vivekktiwari committed:
adding super class to all scale functions

vivekktiwari committed:
add super class to scale funtion

vivekktiwari committed:
add super class to scale functions

vivekktiwari committed:
add super class to scale functions

vivekktiwari committed:
Update continuous_scale.Rd

vivekktiwari committed:
Update discrete_scale.Rd

vivekktiwari committed:
Merge branch 'cran' of https://github.com/faizan-khan-iit/ggplot2 into cran

# Conflicts:
#	DESCRIPTION

vivekktiwari committed:
update describtion

vivekktiwari committed:
for unknown aesthetics and params

vivekktiwari committed:
Update .travis.yml

faizan-khan-iit commented on R/scale-.r:
```diff
@@ -522,7 +522,7 @@ continuous_scale <- function(aesthetics, scale_name, palette, name = waiver(),
                              rescaler = rescale, oob = censor,
                              expand = waiver(), na.value = NA_real_,
-                             trans = "identity", guide = "legend") {
+                             trans = "identity", guide = "legend", super=ScaleContinuous) {
```
@vivekktiwari  Could you explain why we need `ScaleContinuous` as default here?

vivekktiwari replied:
@faizan-khan-iit as per changes in ```ggproto``` (R/ggproto.r), we have a ```super``` class to fetch ```inherited``` class.  (I think ```super``` is being used as an argument for functions if they have a ```ggproto``` call with an inherited function. ) This is was done for: 

```R

-  ggproto(NULL, ScaleContinuous,
+  ggproto(NULL, super,
    call = match.call(),

```

faizan-khan-iit commented on .travis.yml:
```diff
@@ -3,4 +3,4 @@ sudo: false
 cache: packages
 
 only:
-  - cran
```
You should get Travis CI on both the branches..

faizan-khan-iit commented on R/ggproto.r:
```diff
@@ -74,8 +82,17 @@ fetch_ggproto <- function(x, name) {
-    if (is.ggproto(super))
-      res <- fetch_ggproto(super, name)
+    if (is.null(super)) {
+      # no super class
```
@vivekktiwari If there is no super class, do we assign a default class here?

vivekktiwari replied:
@faizan-khan-iit If it is not inheriting any superclass, I think NULL goes as Default automatically as some functions may not have any parent class to catch.

faizan-khan-iit replied:
Then please add a statement explicitly returning `NULL` here along with a comment. But please make sure that is the case before making the changes..

vivekktiwari replied:
@faizan-khan-iit, Please look at the ```res``` variable, I think it is doing the same work, since in ```if``` statement, it does nothing, ```res``` remains ```NULL``` and ignores all other if and else statements and finally at the end it returns ```res``` as ```NULL```

**Please look at the second last line of the code**

```R
fetch_ggproto <- function(x, name) {
  res <- NULL

  val <- .subset2(x, name)
  # The is.null check is an optimization for a common case; exists() also
  # catches the case where the value exists but has a NULL value.
  if (!is.null(val) || exists(name, envir = x, inherits = FALSE)) {
    res <- val
  } else {
    # If not found here, recurse into super environments
    super <- .subset2(x, "super")
    if (is.null(super)) {
      # no super class
    } else if (is.function(super)) {
      res <- fetch_ggproto(super(), name)
    } else {
      stop(
        class(x)[[1]], " was built with an incompatible version of ggproto.\n",
        "Please reinstall the package that provides this extension.",
        call. = FALSE
      )
    }
  }

  res 
#In the end it is return res here, and at the start of the function, **res <- NULL**
}

faizan-khan-iit commented:
@vivekktiwari Could you add more comments in the code describing the changes you made?

faizan-khan-iit commented on R/stat-.r:
```diff
@@ -68,9 +68,9 @@ Stat <- ggproto("Stat",
       c(names(data), names(params)),
-      snake_class(self$stat)
+      snake_class(self)
     )
```
Any particular reason why this is necessary? According to me we should look for layer params in the stat object..

vivekktiwari replied:
@faizan-khan-iit, while trying to find the error, I thought ```stat``` was of no relevance here.... so I removed it, but if I am wrong, I will revert it.

faizan-khan-iit replied:
No need to edit code that works for now. Makes it hard to pinpoint the source of the error. I would suggest reverting it.

faizan-khan-iit commented on R/ggproto.r:
```diff
@@ -43,13 +43,21 @@ ggproto <- function(`_class` = NULL, `_inherit` = NULL, ...) {
-    e$super <- `_inherit`
-    class(e) <- c(`_class`, class(`_inherit`))
-
+    e$super <- find_super
```
@vivekktiwari I am not sure why that error occurs.. May be you should recheck this part of the code.

faizan-khan-iit replied:
I would suggest to put in a few browser calls to make sure we are passing super as function only when required.

vivekktiwari replied:
@faizan-khan-iit I will see how to put browser calls and will do so.

vivekktiwari replied:
I am also lost here, it is hard to understand.

vivekktiwari committed:
Update .travis.yml

vivekktiwari committed:
Update stat-.r

vivekktiwari committed:
Update stat-.r

vivekktiwari committed:
remove one example linked to Hmisc package

vivekktiwari committed:
Update .travis.yml

vivekktiwari commented on R/stat-summary.r:
```diff
@@ -43,7 +43,6 @@
 #' @export
 #' @examples
 #' d <- ggplot(mtcars, aes(cyl, mpg)) + geom_point()
-#' d + stat_summary(fun.data = "mean_cl_boot", colour = "red", size = 2)
```
Removing this example as ```fun.data="mean_cl_boot" ``` is referencing to ```Hmisc```, which is dependent on ```ggplot2```... this may be the cause of error: 

```R
Error in is.list(x) : attempt to apply non-function
Calls: <Anonymous> ... tryCatch -> tryCatchList -> tryCatchOne -> <Anonymous>
```

faizan-khan-iit replied:
I don't think that is the case. Otherwise we would have gotten errors in previous `check()` results. And we have to make sure the tests and examples work fine with ggplot2 loaded as we will need that in the future. So please revert.

vivekktiwari replied:
I tested by removing this particular example, in all condition my ```cran``` check is passing on local machine, but when this example is present, I get the error: 

```R
d <- ggplot(mtcars, aes(cyl, mpg)) + geom_point()
> d + stat_summary(fun.data = "mean_cl_boot", colour = "red", size = 2)
> 
> # You can supply individual functions to summarise the value at
> # each x:
> d + stat_summary(fun.y = "median", colour = "red", size = 2, geom = "point")
Error in stopifnot(is.list(x), is.list(val)) : 
  attempt to apply non-function
Calls: <Anonymous> ... ggplot_build -> <Anonymous> -> f -> <Anonymous> -> stopifnot
```
You can recheck this on your machine, I also looked into ```Hmisc``` package as well, it has ```mean_cl_boot```. that's why when I removed ```ggplot2``` from system, and ran this example, I got: 
```R
> d + stat_summary(fun.data = "mean_cl_boot", colour = "red", size = 2)
Warning message:
Computation failed in `stat_summary()`:
Hmisc package required for this function 
```

I don't know why my commit is not being reflected in ```Travis``` builds, this example is still present there, I just wanted to run the test (just for now by removing this example) so, I can get if this is the sole reason for error, can you please look at the ```Travis logs```? 

Surely, I will revert back, if travis keep on throwing the same error after removing this example.

vivekktiwari replied:
@faizan-khan-iit , Sorry, stupid mistake, forgot to make changes in ```man/stat_summary.Rd``` , ```Travis CI``` is passing on ```cran``` branch, please check.

vivekktiwari replied:
@faizan-khan-iit Your suggestion?

faizan-khan-iit replied:
@tdhock I need a second opinion here. 

As far as I can see, we still need to deal with these issues:

1. Make sure `ggplot2Animint` and `ggplot2` work without errors while being loaded in the same session
2. We could skip this one test that we are failing as this doesn't seem too major with regards to animint code base
3. Make minor changes to documentation in `ggplot2Animint` detailing the purpose of the package and the changes made
4. Merge `animint2` and `ggplot2Animint` into a single package
 
What do you think about this @tdhock? I think we could upload to CRAN after making relevant documentation changes (3). As for (1), (2) and (4), we could work on them afterwards. Any thoughts/suggestions?

faizan-khan-iit replied:
@vivekktiwari  Could you add some details to the README and the package documentation reflecting the changes made till now?

vivekktiwari replied:
@faizan-khan-iit, will update README with the changes made, can you let me know other places that needs to be updated regarding documentation? 

And should we merge this PR now?

faizan-khan-iit replied:
I think minor updates to vignettes should be enough

vivekktiwari replied:
@faizan-khan-iit should I build vignettes and push them too, or just make changes in ```vignettes``` folder as we have ```inst/doc``` in ```.gitignore``` so can't push the builds here?

faizan-khan-iit replied:
No need to make changes to `.gitignore`. But please make sure the vignettes build without errors

vivekktiwari replied:
@faizan-khan-iit, yes, definitely, I ran using ```devtools::build_vignettes()``` and it didn't throw any error, I have pushed changes made in ```vignettes``` folder.

vivekktiwari committed:
Update .travis.yml

vivekktiwari committed:
removing on example on stat_summary

vivekktiwari committed:
Update README.md

vivekktiwari committed:
update README for ggplot2Animint

vivekktiwari committed:
update README for ggplot2Animint

vivekktiwari committed:
update README for ggplot2Animint

vivekktiwari committed:
Merge branch 'cran' of https://github.com/faizan-khan-iit/ggplot2 into cran

vivekktiwari committed:
updating vignettes for ggplot2Animint

tdhock commented:
I would say that (1) is the most important thing to do before a CRAN
submission. I think it is really important that we add some tests to make
sure regular ggplot2 and our code both still work when they are both
loaded/attached.

On Tue, May 29, 2018 at 10:02 AM, Faizan Uddin Fahad Khan <
notifications@github.com> wrote:

> *@faizan-khan-iit* commented on this pull request.
> ------------------------------
>
> In R/stat-summary.r
> <https://github.com/faizan-khan-iit/ggplot2/pull/7#discussion_r191435666>:
>
> > @@ -43,7 +43,6 @@
>  #' @export
>  #' @examples
>  #' d <- ggplot(mtcars, aes(cyl, mpg)) + geom_point()
> -#' d + stat_summary(fun.data = "mean_cl_boot", colour = "red", size = 2)
>
> @tdhock <https://github.com/tdhock> I need a second opinion here.
>
> As far as I can see, we still need to deal with these issues:
>
>    1. Make sure ggplot2Animint and ggplot2 work without errors while
>    being loaded in the same session
>    2. We could skip this one test that we are failing as this doesn't
>    seem too major with regards to animint code base
>    3. Make minor changes to documentation in ggplot2Animint detailing the
>    purpose of the package and the changes made
>    4. Merge animint2 and ggplot2Animint into a single package
>
> What do you think about this @tdhock <https://github.com/tdhock>? I think
> we could upload to CRAN after making relevant documentation changes (3). As
> for (1), (2) and (4), we could work on them afterwards. Any
> thoughts/suggestions?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/faizan-khan-iit/ggplot2/pull/7#discussion_r191435666>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA478h_XZTT5EeNPIrRnvAIedstV0RSHks5t3VT8gaJpZM4UO0Lc>
> .
>

vivekktiwari commented:
@faizan-khan-iit @tdhock but they both have the same function when trying to load/attach them, one will mask the functions from another. (so far I have not got any other error when trying to load/attach both) I'll explain: 

if ```ggplot2``` is loaded and then ```ggplot2Animint```: 

```R
> library(ggplot2)
> library(ggplot2Animint)

Attaching package: ‘ggplot2Animint’

The following objects are masked from ‘package:ggplot2’:

    %+%, %+replace%, aes, aes_, aes_all, aes_auto, aes_q, aes_string, annotate,
    annotation_custom, annotation_logticks, annotation_map, annotation_raster,
    as_labeller, autoplot, benchplot, borders, calc_element, continuous_scale,
    Coord, coord_cartesian, coord_equal, coord_fixed, coord_flip, coord_map,
    coord_munch, coord_polar, coord_quickmap, coord_trans, CoordCartesian,
    CoordFixed, CoordFlip, CoordMap, CoordPolar, CoordQuickmap, CoordTrans,
    cut_interval, cut_number,.......
```
and many more functions, and vice-versa. 

Two Possiblities: 

1. We can put a ```PackageStartMessage("Please detach ggplot2")``` more of as WARNING, when both ```Animint2``` and ```ggplot2Animint``` are being loaded as I don't think users will want to use both packages (```ggplot2``` and ```ggplot2Animint```) at the same time or in the same environment. 

2. We can rename functions, but there are atleast 300+ ```functions/object``` plus, this seems useless as it will not be user-friendly and to make user learn new functions, in place of ```ggplot2``` functions seems inefficient (specially when ```ggplot2``` is more efficient and has more users, problem-solutions), plus, then we will need to change all the functions in ```animint2```, leading to more new syntax. 

This is all I could learn for (1) situation and totally my personal opinion. 

Please Advice!

faizan-khan-iit commented:
@tdhock @vivekktiwari 
I think it is possible for users to want to use both `animint2` and `ggplot2` in the same session. Also by renaming them, we are not defining any new complicated functions. so that should not be a problem as far as I can see. However renaming all the exported objects is not obviously a good idea, and I would like to avoid it too.

A possible solution is to not export all the functions we do not use explicitly. That will narrow down the number of exported functions in `ggplot2Animint` a lot. As for the rest of them, we could safely rename them to animint specific names. For e.g.: `geom_point` gets renamed to `geom_point_animint` / `geom_a_point`. This doesn't require the users to learn any new function names.

@vivekktiwari For now, I suggest you add some explicit tests to make sure `ggplot2` and `ggplot2Animint` do not produce errors while being loaded in the same session.

tdhock commented:
I agree with faizan here. we should probably rename things that we use in a
manner that is consistent easy for ggplot2 users to understand

On Fri, Jun 1, 2018 at 8:32 AM, Faizan Uddin Fahad Khan <
notifications@github.com> wrote:

> @tdhock <https://github.com/tdhock> @vivekktiwari
> <https://github.com/vivekktiwari>
> I think it is possible for users to want to use both animint2 and ggplot2
> in the same session. Also by renaming them, we are not defining any new
> complicated functions. so that should not be a problem as far as I can see.
> However renaming all the exported objects is not obviously a good idea, and
> I would like to avoid it too.
>
> A possible solution is to not export all the functions we do not use
> explicitly. That will narrow down the number of exported functions in
> ggplot2Animint a lot. As for the rest of them, we could safely rename
> them to animint specific names. For e.g.: geom_point gets renamed to
> geom_point_animint / geom_a_point. This doesn't require the users to
> learn any new function names.
>
> @vivekktiwari <https://github.com/vivekktiwari> For now, I suggest you
> add some explicit tests to make sure ggplot2 and ggplot2Animint do not
> produce errors while being loaded in the same session.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/faizan-khan-iit/ggplot2/pull/7#issuecomment-393866323>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA478mkJt-XKD4rLUNbtL_sSUZE3gtUcks5t4TRigaJpZM4UO0Lc>
> .
>

tdhock commented:
I would propose a renaming scheme with the a_ prefix, e.g a_geom_point so
that users can type a_ and then tab-complete

On Fri, Jun 1, 2018 at 8:50 AM, Toby Hocking <tdhock5@gmail.com> wrote:

> I agree with faizan here. we should probably rename things that we use in
> a manner that is consistent easy for ggplot2 users to understand
>
> On Fri, Jun 1, 2018 at 8:32 AM, Faizan Uddin Fahad Khan <
> notifications@github.com> wrote:
>
>> @tdhock <https://github.com/tdhock> @vivekktiwari
>> <https://github.com/vivekktiwari>
>> I think it is possible for users to want to use both animint2 and ggplot2
>> in the same session. Also by renaming them, we are not defining any new
>> complicated functions. so that should not be a problem as far as I can see.
>> However renaming all the exported objects is not obviously a good idea, and
>> I would like to avoid it too.
>>
>> A possible solution is to not export all the functions we do not use
>> explicitly. That will narrow down the number of exported functions in
>> ggplot2Animint a lot. As for the rest of them, we could safely rename
>> them to animint specific names. For e.g.: geom_point gets renamed to
>> geom_point_animint / geom_a_point. This doesn't require the users to
>> learn any new function names.
>>
>> @vivekktiwari <https://github.com/vivekktiwari> For now, I suggest you
>> add some explicit tests to make sure ggplot2 and ggplot2Animint do not
>> produce errors while being loaded in the same session.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/faizan-khan-iit/ggplot2/pull/7#issuecomment-393866323>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AA478mkJt-XKD4rLUNbtL_sSUZE3gtUcks5t4TRigaJpZM4UO0Lc>
>> .
>>
>
>

vivekktiwari commented:
@tdhock @faizan-khan-iit will start looking into it.

vivekktiwari commented:
@faizan-khan-iit while changing functions to ```#' keywords internal``` that are not animint2 specific are failing examples during ```check()```

Example: 

```R
checking examples ... ERROR
Running examples in 'ggplot2Animint-Ex.R' failed
The error most likely occurred in:

> base::assign(".ptime", proc.time(), pos = "CheckExEnv")
> ### Name: aes_position
> ### Title: Position related aesthetics: x, y, xmin, xmax, ymin, ymax, xend,
> ###   yend
> ### Aliases: aes_position x y xmin xmax ymin ymax xend yend
> 
> ### ** Examples
> 
> 
> # Generate data: means and standard errors of means for prices
> # for each type of cut
> dmod <- lm(price ~ cut, data = diamonds)
> cuts <- data.frame(cut = unique(diamonds$cut), predict(dmod, data.frame(cut =
+ unique(diamonds$cut)), se = TRUE)[c("fit", "se.fit")])
> se <- ggplot(cuts, aes(x = cut, y = fit, ymin = fit - se.fit,
+ ymax = fit + se.fit, colour = cut))
> se + geom_pointrange()
> 
> # Using annotate
> p <- ggplot(mtcars, aes(wt, mpg)) + geom_point()
> p + annotate("rect", xmin = 2, xmax = 3.5, ymin = 2, ymax = 25,
+   fill = "dark grey", alpha = .5)
Error in annotate("rect", xmin = 2, xmax = 3.5, ymin = 2, ymax = 25, fill = "dark grey",  : 
  could not find function "annotate"
Execution halted
```

So how should I proceed now?

faizan-khan-iit commented:
If there are any examples that include any of the un-exported functions (like the one above), you should edit them to include `ggplot2Animint:::func_name` to make sure they pass the tests. If they show notes/warnings during the CRAN checks, we will handle them later

vivekktiwari commented:
@faizan-khan-iit, taking one example just to be clear.... 

like in above case. (suppose I have done ```annotate``` and ```annotation_map``` as internal function):

In ``` R/annotation-map.R ```:
```R
#' some description
#' @example
#' # Using annotate
#' p <- ggplot(mtcars, aes(wt, mpg)) + geom_point()
#' p + annotate(............)+annotation_map(.....)
#' @params xxx ....
#' @keywords internal
annotation_map <- function (xxx){

}
```

``` R/annotate.R ```: 
```R
#' some description
#' @params xxx ....
#' @keywords internal
annotate <- function (xxx){

}
```

Solution Suggested: 
```R
#' p <- ggplot(mtcars, aes(wt, mpg)) + geom_point()
#' p + ggplot2Animint:::annotate(............)+ggplot2Animint:::annotation_map(.....)
```
This way, right?

vivekktiwari commented:
[EDITED] Doubt: 
There will be many more functions which are not ```animint2``` specific and have similar examples throughout the code base, not sure if this way I will be to identify which functions should be exported, as ```NAMESPACE``` of ```animint2``` wasn't much help. 
Trick I used: 
removed ```import ggplot2Animint``` from ```animint.R``` and did the ```check()``` threw this error: 

```R
Missing or unexported objects:
  'ggplot2Animint::guides_geom' 'ggplot2Animint::guides_merge'
  'ggplot2Animint::guides_train' 'ggplot2Animint::plot_theme'
  'ggplot2Animint::stairstep'

checking R code for possible problems ... NOTE
animint2dir: no visible global function definition for 'is.ggplot'
make_bar: no visible global function definition for 'stat_summary'
make_bar: no visible global function definition for 'aes_string'
make_tallrect_or_widerect: no visible binding for global variable
  'aes_string'
make_text: no visible global function definition for 'aes_string'
make_text: no visible global function definition for 'geom_text'
scale_size_animint: no visible global function definition for
  'continuous_scale'
Undefined global functions or variables:
  aes_string continuous_scale geom_text is.ggplot stat_summary
````
So, I know these ```function/object``` should be exported for sure. 

Your Suggestions?

vivekktiwari committed:
changing ggplot2 to ggplot2Animint in doc and few doc tweaks

vivekktiwari commented:
typo error, meant `:::`, so sorry.

vivekktiwari committed:
Big commit: change few non-animint specific ggplot2 functions to internal functions

vivekktiwari committed:
wrong test unit

vivekktiwari commented:
@faizan-khan-iit have made some changes as suggested, please review.
Still trying to reduce more export functions.

vivekktiwari commented:
@tdhock your suggestion on how should we step further? 

Also, wanted to ask, can we do something similar like this: [Click Here](https://groups.google.com/forum/#!msg/ggplot2/SZX-SF9UCRo/xxitZ86h1hoJ) for Animint2?

vivekktiwari committed:
Big Commit: Big commit: change few non-animint specific ggplot2 functions to internal functions

vivekktiwari committed:
export draw_key_rect function

tdhock commented:
so actually `:::` is not allowed for non-exported objects on CRAN. instead you can use `get("fun_name", envir = asNamespace("ggplot2Animint"))` ... however these problems could be avoided by simply merging ggplot2Animint into animint2.

vivekktiwari committed:
change in validate_params = FALSE

vivekktiwari committed:
change back validate_params = TRUE

vivekktiwari committed:
Update layer.r

vivekktiwari committed:
Update legend-draw.r

vivekktiwari committed:
Update NAMESPACE

vivekktiwari committed:
Update legend-draw.r

vivekktiwari committed:
renaming ggproto class objects to a_*

vivekktiwari committed:
Update legend-draw.r

vivekktiwari committed:
Merge branch 'cran' into cran

vivekktiwari committed:
Merge branch 'validate-params' into cran

vivekktiwari committed:
Merge pull request #8 from vivekktiwari/cran

rename ggproto class objects to a_* and minor edits

vivekktiwari committed:
export a_draw_key_rect

vivekktiwari committed:
ggplot is now a_plot

vivekktiwari committed:
a_plot in test units instead of ggplot

vivekktiwari commented:
@tdhock @faizan-khan-iit 

I have made following changes here: 

1. `ggplot` is now `a_plot` as suggested in wiki page
2. `ggplot_build` is now `a_plot_build` 
3.  `ggplot_gtable` is now `a_plot_gtable` 

I will try to change as many objects as possible to resolve error when `ggplot2` is loaded alongside `animint2`, once all changes are made here and test cases pass, I will push it to `master` or `merge` branch in `animint2`.  

Update: @faizan-khan-iit @tdhock 

I have also changed all class objects like `Coord`, `Stat`, `Position` to `a_Coord`, `a_Stat`, `a_Position`, hoping this is will also reduce the conflicts in `NAMESPACE` between `animint2` and `ggplot2`

tdhock commented:
sounds great

On Sat, Jun 16, 2018 at 10:38 AM, Vivek Kumar <notifications@github.com>
wrote:

> @tdhock <https://github.com/tdhock> @faizan-khan-iit
> <https://github.com/faizan-khan-iit>
>
> I have made following changes here:
>
>    1. ggplot is now a_plot as suggested in wiki page
>    2. ggplot_build is now a_plot_build
>    3. ggplot_gtable is now a_plot_gtable
>
> I will try to change as many objects as possible to resolve error when
> ggplot2 is loaded alongside animint2, once all changes are made here and
> test cases pass, I will push it to master or merge branch in animint2.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/faizan-khan-iit/ggplot2/pull/7#issuecomment-397816623>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA478nye4KZH7fiGEIQ7xPEqZfZygxa-ks5t9Rh2gaJpZM4UO0Lc>
> .
>

vivekktiwari committed:
all facet_* function are now a_facet_* and add a test unit to ggplot2 compatability

vivekktiwari committed:
+.gg is now +.ggg

vivekktiwari commented:
@faizan-khan-iit I am trying to change `%+%` function, so thought changing its class should help. 
but cran check is failing. 
so I put two `browser()` calls in `extending-ggplot2.Rmd` but they aren't much of a help, output from `browser()` calls below: 
```R
Browse[1]> s
Error in geom_point() + browser() : 
  non-numeric argument to binary operator
debugging in: (function () 
{
    .rs.recordTraceback(FALSE, 2, function(err) {
        .Call("rs_recordNotebookError", err)
    })
})()
debug: {
    .rs.recordTraceback(FALSE, 2, function(err) {
        .Call("rs_recordNotebookError", err)
    })
}
debugging in: ..getNamespace(c("graphics", "3.5.0"), "txhousing")
debug: {
    ns <- .Internal(getRegisteredNamespace(name))
    if (!is.null(ns)) 
        ns
    else tryCatch(loadNamespace(name), error = function(e) {
        warning(gettextf("namespace %s is not available and has been replaced\nby .GlobalEnv when processing object %s", 
            sQuote(name)[1L], sQuote(where)), domain = NA, call. = FALSE, 
            immediate. = TRUE)
        .GlobalEnv
    })
}
```

please suggest what should be my approach?

vivekktiwari commented:
I have commit the changes for better understanding of changes I am making.

vivekktiwari commented:
for some reason, I think it is still calling `+.gg` function from `ggplot2` though it is not loaded but installed.

tdhock commented:
did you try to change the class name from "gg" "ggplot" to something like
"aplot" ?

On Thu, Jun 21, 2018 at 1:55 PM, Vivek Kumar <notifications@github.com>
wrote:

> for some reason, I think it is still calling +.gg function from ggplot2
> though it is not loaded but installed.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/faizan-khan-iit/ggplot2/pull/7#issuecomment-399190503>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA478sxjnfhxVcCRXPalC_5st5_z3SO-ks5t-94SgaJpZM4UO0Lc>
> .
>

vivekktiwari committed:
+.gg is now +.aaa

vivekktiwari commented:
@tdhock your advice helped. thank you! 

I had miss changing `gg` class one place, now changed completely. 
Update: `+.gg` is now `+.aaa` to avoid confusion. 
`a_plot(mpg, aes(displ, hwy))+ggplot2Animint::geom_point()` and
`ggplot(mpg, aes(displ, hwy))+ggplot2::geom_point()` are now working perefectly okay when `library(ggplot2)` is loaded.

vivekktiwari committed:
ggproto func is now a_ggproto

vivekktiwari committed:
exporting few functions

vivekktiwari committed:
export validate_guide

vivekktiwari committed:
export guides_build and minor edits

vivekktiwari committed:
train_layout and build_strip rename with prefix a_

vivekktiwari committed:
validate_params=FALSE

vivekktiwari committed:
export uneval class functions

vivekktiwari committed:
Merge branch 'cran' of https://github.com/faizan-khan-iit/ggplot2 into cran

vivekktiwari committed:
Update CONTRIBUTING.md

vivekktiwari committed:
export more funtions

vivekktiwari committed:
update CONTRIBUTING

Merge branch 'cran' of https://github.com/faizan-khan-iit/ggplot2 into cran

# Conflicts:
#	CONTRIBUTING.md

vivekktiwari committed:
export coord_* functions

vivekktiwari committed:
export a_facet_* functions

vivekktiwari committed:
export fortify function and scale_* functions

vivekktiwari committed:
pushing back to old ggproto function as a_ggproto

vivekktiwari committed:
Update CONTRIBUTING.md

vivekktiwari committed:
change a_geom in rd_aesthetics to geom

vivekktiwari committed:
Merge branch 'cran' of https://github.com/faizan-khan-iit/ggplot2 into cran

vivekktiwari committed:
update layer.r

vivekktiwari committed:
changes in rd_aesthetics, geom_dotplot and minor edits

vivekktiwari committed:
zeroGrob is now a_zeroGrob

vivekktiwari committed:
element is now a_element, all element_* functions are now a_element_*

vivekktiwari committed:
fortify is now a_fortify

vivekktiwari committed:
theme is now a_theme, all theme_* functions are now a_theme_* functions

vivekktiwari committed:
theme is a_theme in test unit

vivekktiwari committed:
scale is now a_scale, all scale_* functions are now a_scale_* functions

vivekktiwari committed:
coord is now a_coord, all coord_* functions are now a_coord_*

vivekktiwari committed:
stat is now a_stat, all stat_* functions are now a_stat_*

vivekktiwari committed:
stat is now a_stat -2

vivekktiwari committed:
geom is now a_geom, all geom_* functions are now a_geom_*

vivekktiwari committed:
position is a_poisition, all position_* functions are now a_poisition_*

vivekktiwari committed:
guide and guides is now a_guide and a_guides, guides_* and guide_* functions are now a_guides_* and a_guide_*

vivekktiwari committed:
all annotation_* functions are now a_annotayion_*

vivekktiwari committed:
all annotation_* functions are now a_annotation_*

vivekktiwari committed:
Merge branch 'cran' of https://github.com/faizan-khan-iit/ggplot2 into cran

vivekktiwari committed:
a_annotate in test units

vivekktiwari committed:
layer is now a_layer, all layer_* functions are now a_layer_*

vivekktiwari committed:
label and labels are now a_label and a_labels, all label_* functions are now a_label_*

vivekktiwari committed:
a_labels in test unit

vivekktiwari committed:
few mean_* functions are now a_mean_*

vivekktiwari committed:
aes and aesthetics are now a_aes and a_aesthetics, all aes_* functions are now a_aes_*

vivekktiwari committed:
removing one test units

vivekktiwari committed:
change in internal function

vivekktiwari committed:
benchplot is now a_benchplot

vivekktiwari committed:
stairstep is now a_stairstep

vivekktiwari committed:
all calc_* functions are now a_calc_*

vivekktiwari committed:
resolution is now a_resolution
